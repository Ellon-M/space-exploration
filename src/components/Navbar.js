import React from 'react';
import { NavLink } from 'react-router-dom';

function Navbar() {
  return (
    <nav className="Navigation-bar">
      <div className="navigation-list">
        <h2>Space Exploration</h2>
        <ul>
          <li><NavLink to="/" className={(navData) => (navData.isActive ? 'active-link' : 'none')}> Rockets </NavLink></li>
          <li><NavLink to="/Missions" className={(navData) => (navData.isActive ? 'active-link' : 'none')}> Missions </NavLink></li>
          <li><NavLink to="/Profile" className={(navData) => (navData.isActive ? 'active-link' : 'none')}> My Profile </NavLink></li>
        </ul>
      </div>
    </nav>
  );
}

export default Navbar;
