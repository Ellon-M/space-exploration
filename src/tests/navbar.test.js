import renderer from 'react-test-renderer';
import { render, fireEvent } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import '@testing-library/jest-dom';
import Navbar from '../components/Navbar';

describe('the navigation bar', () => {
  test('renders', () => {
    const tree = renderer.create(
      <Router>
        <Navbar />
      </Router>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders the rockets page at the /Rockets endpoint and is active', () => {
    const { getByText } = render(
      <Router>
        <Navbar />
      </Router>,
    );
    const rockets = getByText(/Rockets/i);
    fireEvent.click(rockets);
    expect(window.location.pathname).toBe('/');
    expect(rockets.classList.contains('active-link')).toBe(true);
  });

  test('renders the missions page at the /Missions endpoint and is active', () => {
    const { getByText } = render(
      <Router>
        <Navbar />
      </Router>,
    );
    const missions = getByText(/Missions/i);
    fireEvent.click(missions);
    expect(window.location.pathname).toBe('/Missions');
    expect(missions.classList.contains('active-link')).toBe(true);
  });

  test('renders the my profile page at the /Profile endpoint and is active', () => {
    const { getByText } = render(
      <Router>
        <Navbar />
      </Router>,
    );
    const myProfile = getByText(/Profile/i);
    fireEvent.click(myProfile);
    expect(window.location.pathname).toBe('/Profile');
    expect(myProfile.classList.contains('active-link')).toBe(true);
  });
});
