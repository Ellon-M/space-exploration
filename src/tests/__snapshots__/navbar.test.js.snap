// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the navigation bar renders 1`] = `
<nav
  className="Navigation-bar"
>
  <div
    className="navigation-list"
  >
    <h2>
      Space Exploration
    </h2>
    <ul>
      <li>
        <a
          aria-current="page"
          className="active-link"
          href="/"
          onClick={[Function]}
        >
           Rockets 
        </a>
      </li>
      <li>
        <a
          className="none"
          href="/Missions"
          onClick={[Function]}
        >
           Missions 
        </a>
      </li>
      <div
        className="vertical"
      />
      <li>
        <a
          className="none"
          href="/Profile"
          onClick={[Function]}
        >
           My Profile 
        </a>
      </li>
    </ul>
  </div>
</nav>
`;
